python/lsst/analysis/drp/makeQaTractTables.py

```diff
@@ -0,0 +1,326 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import pandas as pd
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+
+__all__ = ["MakeForcedQaTractTablesTaskConfig", "MakeForcedQaTractTablesTask",
+           "MakeUnforcedQaTractTablesTaskConfig", "MakeUnforcedQaTractTablesTask"]
+
+
+class BaseMakeQaTractTablesTaskConnections(pipeBase.PipelineTaskConnections,
+                                           defaultTemplates={"coaddName": "deep"},
+                                           dimensions=("tract", "skymap")):
+    inputObjCats = cT.Input(
+        doc="The coadd catalog from which to collect relevant columns.",
+        storageClass="DataFrame",
+        name="{coaddName}Coadd_obj",
+        dimensions=("tract", "patch", "skymap"),
+        multiple=True,
+        deferLoad=True,
+    )
+    inputObjectTable = cT.Input(
+        doc=("The objectTable_tract associated with the tract.  Only used to ensure the tables persisted "
+             "have the same row ordering as the objectTable_tract tables for ease of joint use."),
+        storageClass="DataFrame",
+        name="objectTable_tract",
+        dimensions=("tract", "skymap"),
+        deferLoad=True,
+    )
+
+
+class BaseMakeQaTractTablesTaskConfig(pipeBase.PipelineTaskConfig,
+                                      pipelineConnections=BaseMakeQaTractTablesTaskConnections):
+    filterMap = pexConfig.DictField(
+        keytype=str, itemtype=str,
+        default={},
+        doc=("Dictionary mapping canonical band to the camera's physical filter.  This is currently "
+             "required for {coaddName}Coadd_obj that are indexed on the latter as the dataId supplied "
+             "will only contain the former (this will likely become unnecessary once the postprocess "
+             "scripts are updated to index on band).")
+    )
+    dfDataset = pexConfig.ChoiceField(
+        dtype=str,
+        default="forced",
+        allowed={
+            "forced": "Measurement values come from the forced_src catalogs",
+            "unforced": "Measurement values come from the meas catalogs",
+        },
+        doc="Type of measurement values of interest, either forced or unforced."
+    )
+    # We want the following to come from the *_meas catalogs as they reflect
+    # what happened in SFP calibration.
+    columnsToCopyFromMeas = pexConfig.ListField(
+        dtype=str,
+        default=["calib_", ],
+        doc=("List of string \"prefixes\" to identify the columns to copy.  All columns with names "
+             "that start with one of these strings will be copied from the *_meas catalogs into the "
+             "*_forced_src catalogs UNLESS the full column name contains one of the strings listed "
+             "in the notInColumnStrList config.")
+    )
```

Comment:
This config option and the one below seem to only be used by the `MakeForcedQaTractTablesTask` subclass, not the `MakeUnforced...` variant.  They should be moved to the config class for just that subclass.